### ISSUE TITLE
[1.16.5 server] Systematic crash if grounded player gives up while being revived

### ISSUE THREAD

**Issue by Watherfoxx** [Cycro54/Revive_Me#19]
**How to reproduce:**
- Kill someone
- Revive him
- He gives up and dies
Whoever was reviving him crashes

**Crash report on the reviver:**

```
Time: 14/12/24 17:02
Description: Ticking entity

java.lang.NullPointerException: Ticking entity
	at net.minecraft.network.PacketBuffer.func_211400_a(PacketBuffer.java:528) ~[?:?] {re:classloading,xf:OptiFine:default,re:mixin,xf:OptiFine:default}
	at net.minecraft.network.PacketBuffer.func_180714_a(PacketBuffer.java:523) ~[?:?] {re:classloading,xf:OptiFine:default,re:mixin,xf:OptiFine:default}
	at invoker54.reviveme.common.network.message.RestartDeathTimerMsg.Encode(RestartDeathTimerMsg.java:27) ~[?:1.6.16] {re:classloading}
	at invoker54.reviveme.common.network.NetworkHandler$$Lambda$13270/60570972.accept(Unknown Source) ~[?:?] {}
	at net.minecraftforge.fml.network.simple.IndexedMessageCodec.lambda$tryEncode$4(IndexedMessageCodec.java:120) ~[?:?] {re:classloading}
	at net.minecraftforge.fml.network.simple.IndexedMessageCodec$$Lambda$15522/273883148.accept(Unknown Source) ~[?:?] {}
	at java.util.Optional.ifPresent(Optional.java:159) ~[?:1.8.0_51] {}
	at net.minecraftforge.fml.network.simple.IndexedMessageCodec.tryEncode(IndexedMessageCodec.java:118) ~[?:?] {re:classloading}
	at net.minecraftforge.fml.network.simple.IndexedMessageCodec.build(IndexedMessageCodec.java:133) ~[?:?] {re:classloading}
	at net.minecraftforge.fml.network.simple.SimpleChannel.encodeMessage(SimpleChannel.java:71) ~[?:?] {re:mixin,re:classloading}
	at net.minecraftforge.fml.network.simple.SimpleChannel.toBuffer(SimpleChannel.java:84) ~[?:?] {re:mixin,re:classloading}
	at net.minecraftforge.fml.network.simple.SimpleChannel.toVanillaPacket(SimpleChannel.java:115) ~[?:?] {re:mixin,re:classloading}
	at net.minecraftforge.fml.network.simple.SimpleChannel.sendTo(SimpleChannel.java:95) ~[?:?] {re:mixin,re:classloading}
	at net.minecraftforge.fml.network.simple.SimpleChannel.sendToServer(SimpleChannel.java:90) ~[?:?] {re:mixin,re:classloading}
	at invoker54.reviveme.client.event.RevivePlayerActionsEvent.reviveCheck(RevivePlayerActionsEvent.java:74) ~[?:1.6.16] {re:classloading}
	at net.minecraftforge.eventbus.ASMEventHandler_228_RevivePlayerActionsEvent_reviveCheck_PlayerTickEvent.invoke(.dynamic) ~[?:?] {}
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-4.0.0.jar:?] {}
	at net.minecraftforge.eventbus.EventBus$$Lambda$3126/1465046657.invoke(Unknown Source) ~[?:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-4.0.0.jar:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-4.0.0.jar:?] {}
	at net.minecraftforge.fml.hooks.BasicEventHooks.onPlayerPreTick(BasicEventHooks.java:71) ~[?:?] {re:classloading}
	at net.minecraft.entity.player.PlayerEntity.func_70071_h_(PlayerEntity.java:196) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
	at net.minecraft.client.entity.player.AbstractClientPlayerEntity.func_70071_h_(AbstractClientPlayerEntity.java:636) ~[?:?] {re:computing_frames,xf:OptiFine:default,xf:fml:forge:bows.0,re:mixin,xf:OptiFine:default,xf:fml:forge:bows.0,re:classloading,xf:OptiFine:default,xf:fml:forge:bows.0,pl:mixin:APP:emotecraft.mixins.json:EmotePlayerMixin,pl:mixin:A}
	at net.minecraft.client.entity.player.ClientPlayerEntity.func_70071_h_(ClientPlayerEntity.java:177) ~[?:?] {re:computing_frames,pl:accesstransformer:B,xf:fml:cmdcam:currentEntity,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,xf:fml:cmdcam:currentEntity,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:cmdcam:currentEntity,pl:mixin:APP:supplementaries.mixins.json:ClientPlayerEntityMixin,pl:mixin:APP:konkrete.mixin.json:client.MixinClientPlayerEntity,pl:mixin:APP:reviveme.mixins.json:LocalPlayerMixin,pl:mixin:APP:mixins.betterthirdperson.json:ClientPlayerEntityMixin,pl:mixin:APP:create.mixins.json:HeavyBootsOnPlayerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.world.ClientWorld.func_217418_a(ClientWorld.java:252) ~[?:?] {re:mixin,xf:OptiFine:default,re:classloading,xf:OptiFine:default,pl:mixin:APP:simplerpc.forge.mixins.json:ClientLevelMixin,pl:mixin:APP:sulfuric.mixins.json:client.world.MixinClientWorld,pl:mixin:APP:mixins.sndctrl.json:MixinClientWorld,pl:mixin:APP:betterbiomeblend.mixins.json:MixinClientWorld,pl:mixin:APP:architectury.mixins.json:MixinClientLevel,pl:mixin:APP:create.mixins.json:BreakProgressMixin,pl:mixin:A}
	at net.minecraft.client.world.ClientWorld$$Lambda$16481/1269750577.accept(Unknown Source) ~[?:?] {}
	at net.minecraft.world.World.func_217390_a(World.java:554) ~[?:?] {re:computing_frames,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:flywheel.mixins.json:TileWorldHookMixin,pl:mixin:A}
	at net.minecraft.client.world.ClientWorld.func_217419_d(ClientWorld.java:210) ~[?:?] {re:mixin,xf:OptiFine:default,re:classloading,xf:OptiFine:default,pl:mixin:APP:simplerpc.forge.mixins.json:ClientLevelMixin,pl:mixin:APP:sulfuric.mixins.json:client.world.MixinClientWorld,pl:mixin:APP:mixins.sndctrl.json:MixinClientWorld,pl:mixin:APP:betterbiomeblend.mixins.json:MixinClientWorld,pl:mixin:APP:architectury.mixins.json:MixinClientLevel,pl:mixin:APP:create.mixins.json:BreakProgressMixin,pl:mixin:A}
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1485) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:simplerpc.forge.mixins.json:ScreenEventsMixin,pl:mixin:APP:multiplayerfix.mixins.json:client.MinecraftMixin,pl:mixin:APP:memoryleakfix-16.mixins.json:targetEntityLeak.Minecraft_targetClearMixin,pl:mixin:APP:konkrete.mixin.json:client.MixinMinecraft,pl:mixin:APP:iceberg.mixins.json:MinecraftMixin,pl:mixin:APP:flywheel.mixins.json:ShaderCloseMixin,pl:mixin:APP:mixins.betterthirdperson.json:MinecraftMixin,pl:mixin:APP:fancymenu.general.mixin.json:IMixinMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:954) [?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:simplerpc.forge.mixins.json:ScreenEventsMixin,pl:mixin:APP:multiplayerfix.mixins.json:client.MinecraftMixin,pl:mixin:APP:memoryleakfix-16.mixins.json:targetEntityLeak.Minecraft_targetClearMixin,pl:mixin:APP:konkrete.mixin.json:client.MixinMinecraft,pl:mixin:APP:iceberg.mixins.json:MinecraftMixin,pl:mixin:APP:flywheel.mixins.json:ShaderCloseMixin,pl:mixin:APP:mixins.betterthirdperson.json:MinecraftMixin,pl:mixin:APP:fancymenu.general.mixin.json:IMixinMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:607) [?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:simplerpc.forge.mixins.json:ScreenEventsMixin,pl:mixin:APP:multiplayerfix.mixins.json:client.MinecraftMixin,pl:mixin:APP:memoryleakfix-16.mixins.json:targetEntityLeak.Minecraft_targetClearMixin,pl:mixin:APP:konkrete.mixin.json:client.MixinMinecraft,pl:mixin:APP:iceberg.mixins.json:MinecraftMixin,pl:mixin:APP:flywheel.mixins.json:ShaderCloseMixin,pl:mixin:APP:mixins.betterthirdperson.json:MinecraftMixin,pl:mixin:APP:fancymenu.general.mixin.json:IMixinMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:184) [?:?] {re:classloading,re:mixin,pl:runtimedistcleaner:A,pl:mixin:A,pl:runtimedistcleaner:A}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51] {}
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:37) [forge-1.16.5-36.2.41.jar:36.2] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider$$Lambda$477/1835777333.call(Unknown Source) [forge-1.16.5-36.2.41.jar:36.2] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-8.1.3.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-8.1.3.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-8.1.3.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82) [modlauncher-8.1.3.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) [modlauncher-8.1.3.jar:?] {}
```

**Comment by Cycro54** [comment]
Ty for the heads up, I'll add the fix before I post the next update

**Comment by Cycro54** [comment]
I had the players String UUID set to null when it should have been an empty string, fixed in revive-me-1.16.5-1.6.24
